src/frequenz/sdk/timeseries/data_pipeline.py

```diff
@@ -0,0 +1,115 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Create, connect and own instances of data pipeline components."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from frequenz.channels import Broadcast, Sender
+
+from frequenz.sdk import microgrid
+
+from ..actor import (
+    ChannelRegistry,
+    ComponentMetricRequest,
+    ComponentMetricsResamplingActor,
+    DataSourcingActor,
+    ResamplerConfig,
+)
+from .ev_charger_pool import EVChargerPool
+from .logical_meter import LogicalMeter
+
+
+@dataclass
+class _ActorInfo:
+    actor: DataSourcingActor | ComponentMetricsResamplingActor
+    channel: Broadcast[ComponentMetricRequest]
+
+
+class DataPipeline:
+    """Create, connect and own instances of data pipeline components."""
+
+    def __init__(
+        self,
+        resampler_config: ResamplerConfig,
+        ev_charger_ids: set[int] | None = None,
+    ) -> None:
+        """Create a `DataPipeline` instance.
+
+        Args:
+            resampler_config: Config to pass on to the resampler.
+            ev_charger_ids: Optional set of IDs of EV Charger to be managed by this
+                instance of the DataPipeline.
+        """
+        self._resampler_config = resampler_config
+        self._ev_charger_ids = ev_charger_ids
+
+        self._channel_registry = ChannelRegistry(name="Data Pipeline Registry")
+
+        self._data_sourcing_actor: _ActorInfo | None = None
+        self._resampling_actor: _ActorInfo | None = None
+        self._logical_meter: LogicalMeter | None = None
+        self._ev_charger_pool: EVChargerPool | None = None
+
+    def _data_sourcing_request_sender(self) -> Sender[ComponentMetricRequest]:
```

Comment:
I'd suggest to put all private methods to the bottom to see the public relevant methods first

Reply:
done.